kips/KIP01 - Prioritize Kolibri Liquidity Pool for Liquidations.md

```diff
@@ -0,0 +1,32 @@
+# Kolibri Improvement Proposal #01
+
+### Preface
+Hello everyone :)  
+I am honored to present the first Kolibri Improvement Proposal. I feel like Satoshi in some sense right now.  
+Kidding aside, I welcome and encourage everyone to participate in this and other proposals.  
+Lets not forget that as kDAO holders and as a community, we are in the control of the future and success of Kolibri.  
+I should also thank the Kolibri dev team for their hardwork and support.
+
+### Summary
+An onchain executable procedure to priorotize the Kolibri Liquidity Pool for oven liquidations.
+
+### Motivation
+The Kolibri Liquidity Pool (KLP) is an important and integral part of the kUSD algorithmic stable coin.  
+It serves at least the following purposes :
+1. Makes sure that big ovens that may not be liquidable by individuals, can be liquidated by the pooling the resources of individual kUSD investors.
+2. Provides a fair return for the investors who are willing to help the kUSD ecosystem stability. 
+
+
+The current implementation doesnt prioritize the KLP and individual investors compete for the liquidation of the under-collaterized ovens using high fees.  
+Such implementation is not healthy for the ecosystem and will discourage the pool investors and may lead to disintegration of KLP due to small or no returns of their investment into KLP.  
+In order to resolve this issue, I propose to give the KLP a X block priority for the liquidations.
```

Comment:
> In general, "time" is hard to measure on the blockchain. I wonder if "time" is the wrong metric. Perhaps we should use collateralization percentage or otherwise? 

Do you mean the 5 block priority time ? the block height is easy to measure on the blockchain which roughly translates to time. although it may not be 100% precise.

>I'm curious about playing with the reward on the liquidity pool.

I dont think this resolves the issue, because the gain to users who liquidate the ovens individually are enormous, compared to anyone participating in the pool and hoping someone else would pay a 1000 XTZ fee to beat the others who are trying to liquidate an oven.
Competing to liquidate the ovens by the fees and hoping miners will pick it up, just does not sit right with me. 

>At a high level - why not just let the KLP always liquidate?

It is a proposal which I think should be considered seriously and be discussed.
From a technical point of view, it is much easier to implement and has less complexity but aside from that, I think letting both private users and KLP have access to Liquidate entry point brings more trust and stability into system.
If giving KLP priority in someway is hard to implement, probably it is the way to go.

Reply:
> At a high level - why not just let the KLP always liquidate?

I think there's real risk to this since with the LP as it is it may fail through a number of mechanisms, potentially destabilizing Kolibri overall. Having individuals still being able to liquidate things means we get to have our cake (prioritize/incentivize the LP) and eat it to (not worry about the negative consequences to the LP being the *only* liquidator). If Quipu doesn't have enough funds or there's some bug in the interaction there it could be catastrophic. 

I do think the reward needs to be higher (probably min 10%, possibly more like 25%), but that's a separate thing.